tools/project_manager/page_detail.php

```diff
@@ -43,18 +47,61 @@
     $round_for_page_selection = null;
 }
 
-$project = new Project($projectid);
 $state = $project->state;
 $title = $project->nameofwork;
 $page_details_str = _('Page Detail');
 
 output_header("$page_details_str: $title", NO_STATSBAR);
 
+// NB We use a flexbox to position the linkbox div rather than just using
+// `float: right` like the other linkboxes in the codebase do because on
+// this page the inflexible and very wide page details table causes the
+// linkbox to display in ugly locations.
+echo "<div style='display: flex; justify-content: space-between'>";
+echo "<div>";
 echo "<h1>" . html_safe($title) . "</h1>\n";
 echo "<h2>$page_details_str</h2>\n";
+echo "</div>";
+if ($may_select_user) {
+    echo "<div>";
+    echo_filter_box($projectid, $show_image_size, $username_for_page_selection, $round_for_page_selection);
+    echo "</div>";
+}
+echo "</div>";
 
 echo "<p>" . return_to_project_page_link($projectid, ["expected_state=$state"]) . "</p>\n";
 
+
+function echo_filter_box(string $projectid, int $show_image_size, ?string $username, ?Round $round)
+{
+    $username = $username === '' ? User::current_username() : $username;
+    $roundid = $round ? $round->id : '';
+    echo "<div id='linkbox'>";
+    echo "<form action='#' method='get'>";
+    echo "<p>";
+    echo "<input type='hidden' name='project' value='$projectid'>";
+    echo "<input type='hidden' name='show_image_size' value='$show_image_size'>";
+    echo "<table>";
+    echo "<tr><td colspan=2>", _("Filter pages by:"), "</td></tr>";
+    echo "<tr>";
+    echo "<td class='bold'>", _("Username"), "</td>";
+    echo "<td><input type='text' name='select_by_user' value='" . attr_safe($username) . "' required></td>";
+    echo "</tr>";
+    echo "<tr>";
+    echo "<td class='bold'>", _("User in round"), "</td>";
```

Comment:
I've added the `<label>`s, but kept the table for now because it's the easiest way to keep the inputs aligned. The received wisdom these days seems to be to use flexboxes, which is fair enough.

I'd rather not make this linkbox a special case for right now though, and I've opened Task #2074 to remind us that we should do this.